pkg/util/test/util.go

```diff
@@ -427,26 +430,41 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
 	if len(kubeconfig) != 0 && kubeconfig[0] != "" {
 		os.Setenv("KUBECONFIG", kubeconfig[0])
 	}
-	cluster, err = operatorops.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
-	if err != nil && !errors.IsNotFound(err) {
-		return err
-	}
-	if cluster.Spec.DeleteStrategy == nil ||
-		(cluster.Spec.DeleteStrategy.Type != corev1.UninstallAndWipeStorageClusterStrategyType &&
-			cluster.Spec.DeleteStrategy.Type != corev1.UninstallStorageClusterStrategyType) {
-		cluster.Spec.DeleteStrategy = &corev1.StorageClusterDeleteStrategy{
-			Type: corev1.UninstallAndWipeStorageClusterStrategyType,
+
+	t := func() (interface{}, bool, error) {
+		cluster, err = operatorops.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+		if err != nil && !errors.IsNotFound(err) {
+			return nil, true, err
 		}
-		if _, err := operatorops.Instance().UpdateStorageCluster(cluster); err != nil {
-			return err
+
+		if cluster.Spec.DeleteStrategy == nil ||
+			(cluster.Spec.DeleteStrategy.Type != corev1.UninstallAndWipeStorageClusterStrategyType &&
+				cluster.Spec.DeleteStrategy.Type != corev1.UninstallStorageClusterStrategyType) {
+			cluster.Spec.DeleteStrategy = &corev1.StorageClusterDeleteStrategy{
+				Type: corev1.UninstallAndWipeStorageClusterStrategyType,
+			}
+
+			if _, err := operatorops.Instance().UpdateStorageCluster(cluster); err != nil {
+				return nil, true, err
+			}
+
+			if err := validateTelemetrySecret(cluster, defaultTelemetrySecretValidationTimeout, defaultTelemetrySecretValidationInterval, false); err != nil {
+				return nil, true, err
+			}
 		}
 
-		if err := validateTelemetrySecret(cluster, defaultTelemetrySecretValidationTimeout, defaultTelemetrySecretValidationInterval, false); err != nil {
-			return err
+		if err := operatorops.Instance().DeleteStorageCluster(cluster.Name, cluster.Namespace); err != nil {
```

Comment:
the deletion actually returns immediately, the storageCluster is still there. we shouldn't need this fix. do you want to use UninstallAndValidateStorageCluster?

Reply:
The method we use for deleting and validating StorageCluster deletion is called `ci_utils.UninstallAndValidateStorageCluster(cluster, t)`, we are already using it and it has
```
	// Delete cluster
	logrus.Infof("Delete StorageCluster %s", cluster.Name)
	err := testutil.UninstallStorageCluster(cluster)
	require.NoError(t, err)

	// Validate cluster deletion
	logrus.Infof("Validate StorageCluster %s deletion", cluster.Name)
	err = testutil.ValidateUninstallStorageCluster(cluster, DefaultValidateUninstallTimeout, DefaultValidateUninstallRetryInterval)
	require.NoError(t, err)
```
The problem there is that some of these APIs are getting connection refused and are not going through, hence why I figured we should add more retries for such calls, I think its better retry some of these than to fail a long running job due to network issues. I am seeing more and more if these last couple days because something going on in the lab. Let me know if you and @pure-jliao want to sync up about this so we can discuss what is the best way to handle these and avoid test failures

Reply:
in that case we can retry for a few times, but we dont have to wait for 30 mins, 3 mins should be good.

Reply:
oh yeah you are right, I will change to 3 mins or less, I missed the point that this is just for deleting and not validating deletion

Reply:
I have changed timeout to 3 mins with 10 second retry interval